tc/core/libraries.h

```diff
@@ -43,6 +43,13 @@ constexpr auto defines = R"C(
 #define inf __longlong_as_double(0x7ff0000000000000LL)
 )C";
 
+constexpr auto warpSyncFunctions = R"C(
```

Comment:
Let's have a comment above saying that "before cuda 9, warps are implicitly synchronized so syncwarp is a noop"